aws/ec2.go

```diff
@@ -0,0 +1,84 @@
+package aws
+
+import (
+	"fmt"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/aws-nuke/logging"
+)
+
+var ec2Instances = make(map[string][]*string)
```

Comment:
I would recommend strongly against stateful variables of this sort. In the future, we'd most likely want to update this code to do work concurrently, and mutable, shared variables will make that very hard.

Instead, the `GetAllResources` function should return this map of EC2 instances and the `nukeAllResources` function should take the map in as an argument. That'll make it easier to test them too. And improve reusability/composability of those functions.